qt/python/mantidqt/plotting/figuretype.py

```diff
@@ -52,8 +56,13 @@ def axes_type(ax):
     # an errorbar plot also has len(lines) > 0
     if len(ax.containers) > 0 and isinstance(ax.containers[0], ErrorbarContainer):
         axtype = FigureType.Errorbar
+        if isinstance(ax, MantidAxes) and ax.is_waterfall():
+            axtype = FigureType.Waterfall
```

Comment:
Just spotted one more thing. It seems it is no longer possible to overplot on waterfall plots by right clicking on the workspace and clicking overplot. I think this will be because it checks if the figure type is a line or error bar plot type so because you have changed waterfall to be a separate type here the check now returns false. I am slightly concerned there are other checks like this that assume the plot is 1d if it has figure type `line` or `errorbar`. (I could be wrong about that, in which case just changing that check would suffice).  Another approach could be to check if the first axes is a waterfall using `ax.is_waterfall()` if the figure is 1D in the `PlotHelpPages`.

Reply:
Good spot, I think this should now be fixed. As far as I can tell this shouldn't create any more issues, as the only other use of FigureType.Line is setting whether the fit button is visible. I was initially performing the check within PlotHelpPages if it returned a 1D plot type, as you suggested. But I felt that this check was outside of the responsiblities of the PlotHelpPages class and the check should be made elsewhere. However, I'm open to moving the check to PlotHelpPages if this could cause issues elsewhere.

Reply:
Ah yes good point, I agree. I'm happy with how it works now then.